monai/apps/auto3dseg/data_analyzer.py

```diff
@@ -162,11 +173,15 @@ def _check_data_uniformity(keys: List[str], result: Dict):
 
         return True
 
-    def get_all_case_stats(self):
+    def get_all_case_stats(self, key="training", transform_list=None):
```

Comment:
There are two risks adding a `transform_list` here:
- A custom `transform_list` can nullify previous settings by the user in the `__init__ `. Docstrings and warnings need to be added.
- DataAnalyzer supports command line interface via Python Fire. But putting a transform_list (list object of Transforms) will be something Python Fire cannot support.